pkg/aws/core.go

```diff
@@ -243,17 +247,59 @@ func (d *Driver) CreateMachine(ctx context.Context, req *driver.CreateMachineReq
 	}
 
 	klog.V(2).Infof("VM with Provider-ID %q, for machine %q should be visible to all AWS endpoints now", response.ProviderID, machine.Name)
+	klog.V(3).Infof("VM with Provider-ID: %q created for Machine: %q", response.ProviderID, machine.Name)
+	return response, nil
+}
+
+// InitializeMachine should handle post-creation, one-time VM instance initialization operations. (Ex: Like setting up special network config, etc)
+// The AWS Provider leverages this method to perform disabling of source destination checks for NAT instances.
+// See [driver.Driver.InitializeMachine] for further information
+func (d *Driver) InitializeMachine(ctx context.Context, request *driver.InitializeMachineRequest) (*driver.InitializeMachineResponse, error) {
+	providerSpec, err := decodeProviderSpecAndSecret(request.MachineClass, request.Secret)
+	if err != nil {
+		return nil, err
+	}
+	instances, err := d.getInstancesFromMachineName(request.Machine.Name, providerSpec, request.Secret)
+	if err != nil {
+		return nil, err
+	}
+	targetInstance := instances[0]
+	providerID := encodeInstanceID(providerSpec.Region, *targetInstance.InstanceId)
+
+	svc, err := d.createSVC(request.Secret, providerSpec.Region)
+	if err != nil {
+		return nil, status.Error(codes.Uninitialized, err.Error())
+	}
 
 	// if SrcAnDstCheckEnabled is false then disable the SrcAndDestCheck on running NAT instance
 	if providerSpec.SrcAndDstChecksEnabled != nil && !*providerSpec.SrcAndDstChecksEnabled {
```

Comment:
```suggestion
	if providerSpec.SrcAndDstChecksEnabled != nil && !*providerSpec.SrcAndDstChecksEnabled && *requiredInstance.SourceDestCheck {
```
No need to call `disableSrcAndDestCheck` if it is already disabled.